app/src/main/java/com/example/onsalestore/MainActivity.java

```diff
@@ -1,14 +1,152 @@
 package com.example.onsalestore;
 
+import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 
+import android.content.Intent;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.MotionEvent;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.example.onsale.R;
+import com.parse.FindCallback;
+import com.parse.LogOutCallback;
+import com.parse.ParseException;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+
+import org.w3c.dom.Text;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class MainActivity extends AppCompatActivity {
+    //UI contents
+    private TextView tvDeals, tvBuyandsell;
+    private ImageView imgMain;
+    private Boolean FirstImageDisplaying = false; //determine if a certain image is displaying
+
+    //categories variables
+    private RecyclerView rvHorizontal;
+    protected CategoryAdapter adapter;
+    protected ArrayList<Categories> allCategories;
+
+    //store variables
+    private RecyclerView rvStoreHorizontal;
+    protected StoresAdapter storesadapter;
+    protected ArrayList<Stores> allStores;
+
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        //UI contents
+        tvDeals = findViewById(R.id.tvDeals);
+        tvBuyandsell = findViewById(R.id.tvBuyandSell);
```

Comment:
Using snake case helps with readability - first letter is lower case and then all other words start with a capital letter.

```suggestion
        tvBuyAndSell = findViewById(R.id.tvBuyandSell);
```